Tests/LibWeb/AccessibilityBridgeMac/tests/test_tree_shape.py

```
@@ -0,0 +1,192 @@
+        self.assertIsNotNone(parent, "AXWebArea's parent must not be null")
+
+    def test_no_ignored_role_node_is_an_accessibility_element(self):
+        """Defense in depth: while walking, every visited element should be a real accessibility element. If an
+        ignored-role node (unnamed paragraph/unnamed generic) ever showed up here, isAccessibilityIgnored regressed —
+        because find_all_by_role / walk only see things AXUIElementCopyAttributeValue surfaces."""
+        offenders = []
+
+        def visit(obj, _depth):
+            role = _ax_attr(obj, kAXRoleAttribute)
+            title = _ax_attr(obj, kAXTitleAttribute)
+            # An AXGroup with no title that has only AXStaticText children is suspicious — would suggest an
```

Ruff (0.15.21)

[warning] Comment contains ambiguous ` ` (NO-BREAK SPACE). Did you mean ` ` (SPACE)?

(RUF003)